### PR TITLE
Tool to suggest smoke tests (in JSON format)

### PR DESCRIPTION
**Pull request by jskeet**
This will make it easier to have confidence in newly-introduced APIs

### CODE REVIEW

File: tools/Google.Cloud.Tools.ReleaseManager/SuggestSmokeTestsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Tools.ReleaseManager
+{
+    /// <summary>
+    /// Command that looks at all the RPCs in a client library and considers which ones might be suitable for smoke tests.
+    /// (This is currently very simplistic, but can become more sophisticated over time.)
+    /// The smoke tests are not written to disk automatically, as they should definitely be checked by hand. For example,
+    /// the Vision API's ProductSearchClient has a PurgeProducts parameterless method - we can't easily tell that that's
+    /// dangerous whereas other parameterless RPCs are safe. (A really sophisticated version of this tool might examine
```
Comment (chrisdunelm) [inline tools/Google.Cloud.Tools.ReleaseManager/SuggestSmokeTestsCommand.cs:33]: You have an amusingly low bar for "really sophisticated" :)

File: tools/Google.Cloud.Tools.ReleaseManager/SuggestSmokeTestsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                }
+                // If there's a method that accepts just a ProjectName, find its equivalent accepting a string. (There should be exactly one.)
+                if (Overloads.Any(o => o.HasSingleParameterOfType("Google.Api.Gax.ResourceNames.ProjectName")))
+                {
+                    var stringParameter = Overloads.Single(o => o.HasSingleParameterOfType(typeof(string))).RequiredParameters[0];
+                    candidate.Arguments[stringParameter.Name] = "projects/${PROJECT_ID}";
+                    return candidate;
+                }
+
+                // If there's a method that accepts just a LocationName, find its equivalent accepting a string. (There should be exactly one.)
```
Comment (chrisdunelm) [inline tools/Google.Cloud.Tools.ReleaseManager/SuggestSmokeTestsCommand.cs:134]: Serious errors here, which need fixing before any further progress can be made in this PR.
There should not be a full stop before the open parenthesis, and the first character in the parenthesis should not be capitalized.  The full stop within the parenthesis should then be moved out of the parenthesis.
Alternatively consider not using parenthesis at all; I enjoy using a semicolon for sentence structures such as this - or possibly a hyphen.

File: tools/Google.Cloud.Tools.ReleaseManager/SuggestSmokeTestsCommand.cs (hunk 1 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager/SuggestSmokeTestsCommand.cs:33]: Maybe "complicated" in this case - we'd have to scan the protos to find the right RPC, then parse the HTTP bindings, which we don't currently use.
A less complex but really, really ugly approach would be to construct the default settings for the RPC and see if it's retryable. Assume we don't want to do anything that can't be retried...

File: tools/Google.Cloud.Tools.ReleaseManager/SuggestSmokeTestsCommand.cs (hunk 2 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager/SuggestSmokeTestsCommand.cs:134]: https://www.youtube.com/watch?v=kfVsfOSbJY0

**Pull request merged**